IPython/terminal/interactiveshell.py

```diff
@@ -448,24 +474,17 @@ class to use for the `NavigableAutoSuggestFromHistory` to request
         `stream_inline_completions`
     """,
     ).tag(config=True)
+    _llm_provider_class: Any = None
 
     @observe("llm_provider_class")
     def _llm_provider_class_changed(self, change):
         provider_class = change.new
-        if provider_class is not None:
-            warn(
-                "TerminalInteractiveShell.llm_provider_class is a provisional"
-                "  API as of IPython 8.32, and may change without warnings."
-            )
-            if isinstance(self.auto_suggest, NavigableAutoSuggestFromHistory):
-                self.auto_suggest._llm_provider = provider_class()
-            else:
-                self.log.warn(
-                    "llm_provider_class only has effects when using"
-                    "`NavigableAutoSuggestFromHistory` as auto_suggest."
-                )
+        self._llm_provider_class = provider_class
```

Comment:
Should we keep the provisional warning on backport?

```suggestion
        if provider_class is not None:
            warn(
                "TerminalInteractiveShell.llm_provider_class is a provisional"
                "  API as of IPython 8.32, and may change without warnings."
            )
        self._llm_provider_class = provider_class
```